### ISSUE TITLE
Support for nested fragments

### ISSUE THREAD

**Issue by aleccool213** [dotansimha/graphql-eslint#219]
When using the tool I noticed that nested fragments are not supported and throw an error:

```graphql
fragment price on Price {
  value
  currency
}

fragment example on UnitVariant {
  id
  price {
    ...price
  }
}
```

<img width="834" alt="Screen Shot 2020-12-15 at 8 02 32 AM" src="https://user-images.githubusercontent.com/6475934/102239715-fffc8100-3eab-11eb-95c1-c102363f5443.png">

Is this intentional? Something we can support?

**Comment by dotansimha** [comment]
Yeah, totally. I think we'll need to improve support for that. It's might be tricky since we need to detect those (related to the siblings operation we are working on now)


In the meantime, you can disable this specific validation rule:

```
'@graphql-eslint/validate-against-schema': ['error', {
  disableRules: ["KnownFragmentNamesRule"]
}]
```

**Comment by dotansimha** [comment]
All graphql-js validations will be standalone rules soon, so it should make it simpler to use custom rules.

**Comment by dontsave** [comment]
We have noticed this issue also affects the `require-id-when-available` rule, where ids will be on nested fragments, but the linter flags them as not there

**Comment by B2o5T** [comment]
Hi there, does anyone still have this problem with `known-fragment-names` or `require-id-when-available` rule? I can't make a reproduction of this problem on the last version of `graphql-eslint`

**Comment by B2o5T** [comment]
Closing due to inactivity. Please feel free to reopen

**Comment by joeboyscout04** [comment]
> Hi there, does anyone still have this problem with `known-fragment-names` or `require-id-when-available` rule? I can't make a reproduction of this problem on the last version of `graphql-eslint`

Sorry, missed this message.  I think we still have a problem with `known-fragment-names`.  We have many graphQl files, and sometimes the fragments are not in the same file as the operations.   This produces errors if we use `known-fragment-names`, but the graphQl queries function fine in our codebase. 

Is this something covered by the existing issue?

**Comment by B2o5T** [comment]
Hi @joeboyscout04,
What is version of `graphql-eslint` are you using?
How do you import fragments in operations that are not in the same file?

**Comment by B2o5T** [comment]
Did you use [#import comments](https://github.com/dotansimha/graphql-eslint/blob/master/docs/rules/known-fragment-names.md#correct-existing-import-to-userfields-fragment) while the import fragments from another file?

Otherwise, if you use `#import comments` and it's not working for you please make a reproduction repo and I'll help you.

**Comment by joeboyscout04** [comment]
Excellent, I’ll give that a try. 

- Joseph Elliott
Developing apps for a better world.

**Comment by joeboyscout04** [comment]
Worked great.  Thanks for the help, we can close this issue again @B2o5T 👍
